include/mlboard_logger.hpp

```diff
@@ -0,0 +1,200 @@
+/**
+ * @file mlboard_logger.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of the mlbaord logger callback function.
+ */
+#ifndef ENSMALLEN_CALLBACKS_MLBOARD_LOGGER_HPP
+#define ENSMALLEN_CALLBACKS_MLBOARD_LOGGER_HPP
+
+#include <ensmallen.hpp>
+#include <mlboard/core.hpp>
+#include <mlboard/mlboard.hpp>
+
+namespace ens {
+
+/**
+ * Mlboard Logger function, based on the EndEpoch callback function.
+ */
+class MlboardLogger
+{
+ public:
+  /**
+   *
+   * @param output Filewriter object to log the metrics.
+   * @param epochCount Interval of epochs youw want to log your data.
+   * @param accTag Tag to use for accuracy.
+   * @param lossTag Tag to use for loss
+   */
+  MlboardLogger(
+      mlboard::FileWriter& output,
+      int epochCount = 1,
+      std::string accTag = "accuracy",
+      std::string lossTag = "loss") : callbackUsed(false),
+      output(output),
+      summaryType(""),
+      epochCount(epochCount),
+      accTag(accTag),
+      lossTag(lossTag)
+  { /* Nothing to do here. */ }
+
+  /**
+   *
+   * @param output Filewriter object to log the metrics.
+   * @param func A custom function to log some metrics, which return the
+   *    value to be logged.
+   * @param epochCount Interval of epochs youw want to log your data.
+   * @param accTag Tag to use for accuracy.
+   * @param lossTag Tag to use for loss.
+   */
+  MlboardLogger(
+      mlboard::FileWriter& output,
+      std::function<double()> func,
+      int epochCount = 1,
+      std::string accTag = "accuracy",
+      std::string lossTag = "loss")
+    : callbackUsed(true),
+      output(output),
+      summaryType(""),
+      localFunc(func),
+      epochCount(epochCount),
+      accTag(accTag),
+      lossTag(lossTag)
+  {
+    // Nothing to do here.
+  }
+
+  /**
+   *
+   * @param output Filewriter object to log the metrics.
+   * @param func A custom function to log some metrics, which return the
+   *    value to be logged.
+   * @param epochCount Interval of epochs youw want to log your data.
+   * @param summaryTag Tag to use for summary.
+   * @param summaryType Type of summary to be logged.
+   * @param embeddingMetadata Metadata for embedding.
+   * @param heighofImage The height of image to be logged.
```

Comment:
I think a better name is `imageHeight` and `imageWidth`. Also there is a typo on this line.

Reply:
Done